isthmus/src/main/java/io/substrait/isthmus/SubstraitRelNodeConverter.java

```diff
@@ -48,57 +50,67 @@
  */
 public class SubstraitRelNodeConverter extends AbstractRelVisitor<RelNode, RuntimeException> {
 
-  private final RelOptCluster relOptCluster;
-  private final CalciteCatalogReader catalogReader;
-
-  private final SimpleExtension.ExtensionCollection extensions;
+  private final RelDataTypeFactory typeFactory;
 
   private final ScalarFunctionConverter scalarFunctionConverter;
-
   private final AggregateFunctionConverter aggregateFunctionConverter;
   private final ExpressionRexConverter expressionRexConverter;
 
   private final RelBuilder relBuilder;
 
+  public SubstraitRelNodeConverter(
+      SimpleExtension.ExtensionCollection extensions,
+      RelDataTypeFactory typeFactory,
+      RelBuilder relBuilder) {
+    this.typeFactory = typeFactory;
+    this.relBuilder = relBuilder;
+
+    this.scalarFunctionConverter =
+        new ScalarFunctionConverter(extensions.scalarFunctions(), typeFactory);
+
+    this.aggregateFunctionConverter =
+        new AggregateFunctionConverter(extensions.aggregateFunctions(), typeFactory);
+
+    this.expressionRexConverter =
+        new ExpressionRexConverter(
+            typeFactory, scalarFunctionConverter, aggregateFunctionConverter);
+  }
+
+  @Deprecated
   public SubstraitRelNodeConverter(
```

Comment:
I would opt to remove this constructor entirely, but I have not at this point because I believe that would be a breaking change. As such I have marked is as deprecated for now.

Reply:
I'm okay with removing this. It's a breaking change but we're still earlier. Just add a breaking change note to the PR description

Reply:
Went ahead and remove this constructor. Rebased to make the history nicer as well.